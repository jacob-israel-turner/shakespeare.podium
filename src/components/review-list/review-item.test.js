import React from 'react'
import renderer from 'react-test-renderer'
import ReviewItem from './review-item'
import {StyleSheetTestUtils} from 'aphrodite'
import {shallow} from 'enzyme'

beforeAll(() => {
  StyleSheetTestUtils.suppressStyleInjection()
});

afterAll(() => {
  StyleSheetTestUtils.clearBufferAndResumeStyleInjection()
});

describe('<ReviewItem />', () => {
  const fakeReview = {
    onClick: jest.fn(),
    author: 'Jacob Turner',
    rating: 4.3,
    publish_date: new Date('01/12/17')
  }

  afterEach(() => fakeReview.onClick.mockReset())

  it('renders without crashing', () => {
    const component = renderer.create(<ReviewItem {...fakeReview} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  });

  it('renders its given props', () => {
    const component = renderer.create(<ReviewItem {...fakeReview} />)
    const tree = component.toJSON()
    expect(tree.children.length).toBe(3)
    const [name, rating, date] = tree.children.map(({children}) => children[0]);
    expect(name).toBe(fakeReview.author)
    expect(rating).toBe(fakeReview.rating)
    expect(new Date(date).getTime()).toBe(new Date(fakeReview.publish_date).getTime())
  })

  it('invokes its callback', () => {
    const component = shallow(<ReviewItem {...fakeReview} />)
    component.simulate('click')
    expect(fakeReview.onClick).toHaveBeenCalledTimes(1)
  })
})
