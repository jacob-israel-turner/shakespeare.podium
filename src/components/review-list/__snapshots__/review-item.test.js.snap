exports[`<ReviewItem /> renders without crashing 1`] = `
<tr
  className="container_120drhm"
  onClick={[Function]}>
  <td>
    Jacob Turner
  </td>
  <td>
    4.3
  </td>
  <td>
    01/12/2017
  </td>
</tr>
`;
